internal/store/fake_store_test.go

```diff
@@ -400,29 +401,62 @@ func TestFakeStoreService(t *testing.T) {
 	assert.Nil(service)
 }
 
-func TestFakeStoreEndpiont(t *testing.T) {
-	assert := assert.New(t)
-	require := require.New(t)
-
-	endpoints := []*corev1.Endpoints{
+func TestFakeStoreEndpointSlice(t *testing.T) {
+	t.Parallel()
+	endpoints := []*discoveryv1.EndpointSlice{
 		{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      "foo",
+				Name:      "foo-1",
 				Namespace: "default",
+				Labels: map[string]string{
+					discoveryv1.LabelServiceName: "foo",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-1",
+				Namespace: "bar",
+				Labels: map[string]string{
+					discoveryv1.LabelServiceName: "foo",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-2",
+				Namespace: "bar",
+				Labels: map[string]string{
+					discoveryv1.LabelServiceName: "foo",
+				},
 			},
 		},
 	}
-	store, err := NewFakeStore(FakeObjects{Endpoints: endpoints})
-	require.Nil(err)
-	require.NotNil(store)
-	c, err := store.GetEndpointsForService("default", "foo")
-	assert.Nil(err)
-	assert.NotNil(c)
 
-	c, err = store.GetEndpointsForService("default", "does-not-exist")
-	assert.NotNil(err)
-	assert.True(errors.As(err, &ErrNotFound{}))
-	assert.Nil(c)
+	store, err := NewFakeStore(FakeObjects{EndpointSlices: endpoints})
+	require.Nil(t, err)
+	require.NotNil(t, store)
+
+	t.Run("Get EndpointSlices for Service with single EndpointSlice", func(t *testing.T) {
+		t.Parallel()
+		c, err := store.GetEndpointSlicesForService("default", "foo")
+		assert.Nil(t, err)
```

Comment:
```suggestion
		require.NoError(t, err)
```

No point in continuing when `err != nil`. The same elsewhere.